test/export/test_export.py

```diff
@@ -870,7 +876,7 @@ def forward(self, x):
                 # z = 4
                 return x + y + z + w2
 
-        ep = export(M(), (torch.randn(2, 3),), strict=False)
+        ep = export(M(), (torch.randn(2, 3),), strict=False).run_decompositions({})
```

Comment:
Why these extra `run_decompositions`? I guess I'm missing context on this PR, what does it mean to make these tests "training IR compatible"? Please add a summary to the PR.

Reply:
The test expects to see mutated buffer as extra output. Only way we get it today is via functiionalizaiton.